### ISSUE TITLE
Testing commits of sergiomb2/rawstudio

### ISSUE THREAD

**Issue by eduardtibet** [sergiomb2/rawstudio#1]
On behalf of https://github.com/rawstudio/rawstudio/issues/46#issuecomment-359302737
@sergiomb2 Hmm... I'm just a novice user of rawstudio (neither C developer, nor digital photographer).

As fo compiling - yes, it compiled and installed. If you let me know, what function should I look at - I will do it. But by now really I don't know where to look at :)

BTW, I've already have RS 2.0 installed with no system theme by default and after installing 2.1 it continues to use this setting - not system theme. But i suggest it is because of local settings from 2.0

**Comment by eduardtibet** [comment]
Here are some output after running:

```
(rawstudio:24184): Gtk-WARNING **: Allocating size to rawstudio-widget 0x561667cee270 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.

(rawstudio:24184): Gtk-CRITICAL **: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar
```

**Comment by esbante** [comment]
I have compiled it now and it is running smoothly. Can't really tell any difference from the "master" version though. The camera support seems to be the same.

**Comment by esbante** [comment]
I've had time to download and test various raw files now and this version seems to have excellent support, even for cameras as new as a7r mk3. Only problem is to find dcp-profiles as these are (of course) not included. The default white balance was off in the Nikon d850 files, but works fine in all the other camera files i've tested. Running smoothly. Great work!

**Comment by sergiomb2** [comment]
Thank you for your feedback,  we have this build in fedora updates-testing [1] 

[1]
https://bodhi.fedoraproject.org/updates/?packages=rawstudio

**Comment by esbante** [comment]
It crashes when trying to open Fuji .RAF-files. But I have never managed to edit them in rawstudio before either. Darktable handles .RAF files well though, so I guess it should be possible?

**Comment by sergiomb2** [comment]
the crash is with plugin rawspeed or with plugin ufraw ? can you please send me one sample

**Comment by esbante** [comment]
With rawspeed. I just built the devel branch. Will test the dcraw one shortly. It crashes with files from the Fujifilm x-t2, files from the x-t1 are loaded into the program but there are lines across the image and the colors are strange.

The files are too big for attaching here. You can for example download a sample from here (x-t2): https://www.imaging-resource.com/PRODS/fuji-x-t2/Y-AE-0297.RAF.HTM

And here (x-t1):

https://www.imaging-resource.com/PRODS/fuji-x-t1/Y5-100-400-118.RAF.HTM

**Comment by esbante** [comment]
The dcraw branch works equally good for me, but it also crashes when trying to load .raf files.

**Comment by esbante** [comment]
I was too quick in my judgement. I am not able to export the files in either one of the branches unfortunately.

**Comment by sergiomb2** [comment]
you mean , export doesn't work , in both branches devel and load-dcraw  ?  , ok  but no need test load-dcraw since devel have it also, load-dcraw branch if just to do one pull request to rawstudio project. 
dcraw crash with images raf , maybe you found a bug in original code, I will try check it in this weekend. 
 
Many thanks

**Comment by esbante** [comment]
Excactly, export doesn't work for me in load-dcraw or devel. And yes, I think the problem with .raf files has been there for a long time.

Thank you for working on this project!

**Comment by sergiomb2** [comment]
OK
load-dcraw is one copy of ufraw and ufraw is based on dcraw , but all these projects may be replace with libraw [1] . Maybe that is why dcraw and ufraw looks like they are abandoned , so maybe I should try switch to LibRaw ... , rawspeed also moved to have an API and be compiled as one library. 

Many thanks, 

[1]
LibRaw is based on the source codes of the dcraw utility, where part of drawbacks have already been eliminated and part will be fixed in future. The users of the library are provided with API to be built into their software programs.

**Comment by esbante** [comment]
Yes, sounds wise. I uninstalled and recompiled. Now the export works fine in all versions. Same issue with .raf files though.

**Comment by rnwgnr** [comment]
I tried to build the devel-branch on Manjaro, but there is a header missing:
```
rawstudio-plugin-api.cpp:20:10: fatal error: StdAfx.h: No such file or directory
 #include "StdAfx.h"
          ^~~~~~~~~~
compilation terminated.
```
Unfortunately all i can find about this header refers with Visual Studio...any hint what needs to be done to build this with manjaro?

**Comment by sergiomb2** [comment]
it is in plugins/load-rawspeed/rawspeed/RawSpeed/StdAfx.h
did you pull submodule "plugins/load-rawspeed/rawspeed"  ?

**Comment by rnwgnr** [comment]
I obviously missed that one :)
Build is now fine, i'll come back later for feedback.

**Comment by rnwgnr** [comment]
Hi again,

did some tests and everything seems to work fine. Good work. :+1: 
Would like to see this merged in the official repo.

**Comment by esbante** [comment]
Hello again. Just installed Ubuntu 18.04, and I have compiled rawstudio but it refuses to run. Not compatible with 18.04?

**Comment by sergiomb2** [comment]
it runs on Fedora 28 , can you give us any clue ?

**Comment by esbante** [comment]
When running ./autogen.sh (after resolving dependencies) I get the following message: 

Please add the files
  codeset.m4 gettext.m4 glibc21.m4 iconv.m4 isc-posix.m4 lcmessage.m4
  progtest.m4
from the /usr/share/aclocal directory to your autoconf macro directory
or directly to your aclocal.m4 file.
You will also need config.guess and config.sub, which you can get from
ftp://ftp.gnu.org/pub/gnu/config/.

It was possible to make and install it, but the program would not open. I managed to find all *.m4-files except for isc-posix-m4 and moved them to the autoconf directory, and now it is working.

**Comment by esbante** [comment]
But it left me with a different problem: the vignette tool does not work for any camera files, old or new, not even after updating the lensfun data.

**Comment by esbante** [comment]
I tried building it again, but now it won't start. Likewise with the main rawstudio version.

**Comment by sergiomb2** [comment]
hi @esbante  Fedora F26, F27 , F28 and rawhide have this version (git master)

**Comment by esbante** [comment]
Ok, thanks, I will try Fedora instead then.

**Comment by esbante** [comment]
I have installed fedora 28 and rawstudio from the program app. It was probably wayland that caused the crashing, because it crashed after installation in fedora 28 too. When choosing gnome x-org or gnome classic in the login it did not crash. The vignette tool still does not work however. What version of fedora do you use, and do you use gnome?

**Comment by sergiomb2** [comment]
:P sorry I didn't notice your second reply , please open a new issue just about "vignette tool".
Yes unfortunately I hadn't time to check core dump on wayland , I use Fedora 27 KDE ... 
Thanks.

**Comment by esbante** [comment]
Hi! Sorry for not writing, I have been away for a while. I tried Fedora 28 and rawhide, and as long as I used Gnome 3 it crashed, with KDE, XFCE and so on it did not crash. In Ubuntu 18.04 the issue remained the same, crashes in Gnome 3, but not in Gnome 2 etc. I could not get the vignette tool to work however, and the layout looked a bit awkward, so I switched to Ubuntu 16.04 with Unity. Now I have a working vignette tool and no crashes. I do not remember, but I think I had to run ldconfig to get it to work properly ( I am not very knowledgable when using Linux, I just read here and there and try to get things to work, and I think that was necessery). In the fork by jbuchbinder he has made a commit which says  "Fix compilation on modern Ubuntu.", so maybe there is a compatibility problem.

**Comment by sergiomb2** [comment]
@esbante rawstudio crash with xwayland backend with gnome 3 etc try run:
`export GDK_BACKEND=x11; rawstudio`

**Comment by esbante** [comment]
Ok, will try that. Can you edit .raf files from fuji cameras on your installation of rawstudio? For me that has never been possible.

**Comment by sergiomb2** [comment]
@esbante , I tested [1] , I can view Y5-100-400-118.RAF (modify seems not want to work) but Y-AE-0297.RAF core dumped . 
So I test directly with ufraw [2] no problem ( I can modify the raw) but with [3] also core dumped , so if ufraw itself fails, rawstudio can't do better IMHO.

BTW: In the fork by jbuchbinder he has made a commit which says "Fix compilation on modern Ubuntu." It can't fix anything , just check the existence of one command , that help on translations. 

Also btw feature "light out" breaks my kde on Fedora 27 , until I kill rawstudio .

[1]   
```
ll -h
total 80M
-rw-rw-r-- 1 sergio sergio 32M set 15 20:05 Y5-100-400-118.RAF
-rw-rw-r-- 1 sergio sergio 48M set 15 20:05 Y-AE-0297.RAF
```
[2]
ufraw Y5-100-400-118.RAF

 [3] 
ufraw Y-AE-0297.RAF
Segmentation fault (core dumped)

**Comment by esbante** [comment]
Hi! It's been a while, but I have encountered some problems with rawstudio on newer distributions. First of all I would like to thank you for the exiv2 commit! As I mentioned earlier I got rawstudio to work in distributions based on Ubuntu 16.04. In newer distributions (I have tested Fedora 30 KDE, Ubuntu 18.04 and all flavours of Manjaro) the problems are the same. With cameras that have an original rawstudio profile there are no problems. I have to set a lens (for example generic rectilinear 10-1000) to get the vignette tool to work, which is not necessary in older distribution, so that is working even though it is different. The problem is that the adobe standard profile makes over exposed areas black. If I use the camera profiles called for example natural etc from adobe camera raw the over exposed areas get alot of artifacts, especially the color red. In Ubuntu 16.04 those problems do not exist. Older cameras with support in the "original" rawstudio (those who have a rawstudio simple profile) work as they should. What could be the cause of this? Something caused by rawspeed, or just the profiles used? I tried to use profiles from rawtherapee, but that was much worse.

**Comment by sergiomb2** [comment]
Thanks for report 
in devel branch, I had update ufraw to 9.28 commit "update plugins/load-dcraw (ufraw) with dcraw_9.28" many could fix something 
in Ubuntu 16.04 you build the same sources ? , maybe could be because now we have enable openmp ...

**Comment by esbante** [comment]
Yes I did build the devel branch, but that was some time ago so maybe things have changed since. I will rebuild it in 16.04 and report back.

**Comment by sergiomb2** [comment]
```
BuildRequires: 
gcc-c++ 
gtk3-devel 
libxml2-devel 
GConf2-devel 
dbus-devel 
lcms2-devel 
libjpeg-devel 
libtiff-devel 
exiv2-devel 
lensfun-devel 
fftw-devel 
sqlite-devel 
gphoto2-devel 
desktop-file-utils
```

these are the package that we use to build rawstudio , what package could break the colors ? lcms2 ? gPhoto2 ?

**Comment by esbante** [comment]
Lcms2 maybe? Or could it be something regarding rawspeed and the color profiles? What I find strange is that the cameras supported in the "original" rawstudio, cameras with a built in simple color profile work as they should.

**Comment by sergiomb2** [comment]
@esbante 

> Yes I did build the devel branch, but that was some time ago so maybe things have changed since. I will rebuild it in 16.04 and report back.

if rawstudio (rebuild) in 16.04  works without the problems that you have on 18.x , we need to explore what 3rd party software update break rawstudio ...

**Comment by esbante** [comment]
I have built it in 16.04 and it works just like earlier versions of rawstudio. To clarify: The difference between Ubuntu 16.04 and 18.04 (and all recent releases of fedora, majaro etc) is that the adobe standard profile for cameras without an original rawstudio profile shows the overexposed areas as black in the newer releases. In Ubuntu 16.04 the overexposed areas are shown as black until I make a click on the curve, then everything goes back to normal. I did however discover that the camera profiles called natural etc. from adobe camera raw that I thought worked ok in 16.04 do not. The issues there are the same as in 18.04, color artifacts when overexposed and the shadows turn grey when pushed to black. I have found a picture profile for another modern camera (Sony a7r2) that works without above issues, so it seems to have something to do with the color profiles and how they clip the colors. Lcms2 is the package connected to icc-profiles right?

**Comment by sergiomb2** [comment]
maybe related with https://github.com/rawstudio/rawstudio/issues/7

**Comment by esbante** [comment]
No, I do not think it is connected to that. I have tried with several distros on several different computers, and the issue is the same. Weird color clipping issues on cameras that were not supported in the last official rawstudio version (all cameras with built in color profiles). If it is caused by lcms2 version 2.9, how do we test that?

**Comment by esbante** [comment]
When using other profiles on cameras with working built-in profiles, they also show the same clipping issues. So, color profiles that work on older linux distributions (without issues in ubuntu 14.04, some acceptable issues in ubuntu 16.04, and unacceptable issues in 18.04 and onwards (recent fedora, manjaro etc included)) do not work in the more recent ones.

**Comment by esbante** [comment]
I have investigated the matter further (it is the same in this devel branch as in the main rawstudio/rawstudio branch) and I have now found a working solution for my Pentax k-1. I extracted the embedded profile from the dng using adobe profile editor, and that profile works without issues. I tried modifying the adobe standard profile, but it did not work. I have checked some other cameras briefly as well, and most of them seem to accept dcp-profiles from the camera folder in adobe camera raw, but none of them accepts the adobe standard profile without clipping the colors (the images look like you have the exposure mask activated). All icc-profiles seem to work, but camera specific icc-profiles are quite difficult to find, and while adjusting the white balance works, it always looks like it is set to 5000/0 to begin with, even when clicking for custom white balance in the image, it can be adjusted from that starting point however.

So, to conclude, adobe standard dcp profiles are not compatible with rawstudio in recent linux editions. A library of working dcp profiles would be nice to add to the repository, but I do not know if it is legal to add the profiles from adobe camera raw or if we would have to create our own? ICC-profiles are working, with some limitations in the interface.

**Comment by esbante** [comment]
I have tried to install the old version of lcms2 in newer linux editions, but I have not succeded, so I do not know if it is that or something else that is causing this problem.

**Comment by sergiomb2** [comment]
Thanks for the report , is good to know that I haven't break anything ;) [1]
About ICC-profiles, dcp-profiles , can you send me that profiles that you mention , I do not know what is  ICC-profiles and dcp-profiles , can you elucidate me where or how I found folder mention on "most of them seem to accept dcp-profiles from the camera folder in adobe camera raw ? 

Thanks

[1]
it is the same in this devel branch as in the main rawstudio/rawstudio branch

**Comment by esbante** [comment]
This explains it pretty well: https://rawstudio.org/blog/?p=236

So, without a dcp or icc profile the image in rawstudio shows up as green. Dcp profiles are camera specific, but icc profiles can be used with any camera in rawstudio (but there will be problems with getting the colors right if it is not camera specific). There are profiles made by the rawstudio developers in the profiles folder of rawstudio. For more recent cameras the easiest way is to install adobe dng converter (free download) in wine or on a windows computer, and then you can find dcp profiles at C:\ProgramData\Adobe\CameraRaw\CameraProfiles\Camera\ or C:\ProgramData\Adobe\CameraRaw\CameraProfiles\adobe standard. None of the profiles in the "adobe standard" folder seem to work in recent linux releases, but the ones in the the "camera" folder seem to work with most cameras (not the pentax k-1 though for example). There are other sources for dcp profiles, such as from rawtherapee, but these do not seem to work totally ok, or (and these seem to work) for Sony cameras at for example http://www.piraccini.net/2011/02/profili-colore-sony-a900-per-adobe-lr.html. There is also the option to buy profiles from huelight, but these didi not work on my pentax k-1 at least. You can also make your own profile with the adobe program, but it doesn't work on linux, or for linux dcamprof or other programs. But that involves some work and color charts etc.

The easiest way would perhaps be to edit the adobe profiles in adobe profile editor and include them into the profiles folder of rawstudio.

I have attached the working profile for Pentax k-1 that I extracted from the dng.

[PENTAX K-1 embedded.zip](https://github.com/sergiomb2/rawstudio/files/3743300/PENTAX.K-1.embedded.zip)

**Comment by esbante** [comment]
I guess it is better to create own profiles to not get in trouble with copyright etc., but I do not know what the status of the free adobe profiles are.

**Comment by esbante** [comment]
I am afraid the embedded profile did not work either, blue often turns out purple and overexposed areas sometimes turn pink. I will revert to some kind of ubuntu 16.04 based distribution and use that from now on.

**Comment by esbante** [comment]
Would it be possible to make an appimage or snap from this? If made in the Trusty Tahr 14.04 environment I guess it would work perfectly, and it would work very good in 16.04. I have tried to do this but I can not figure out how. Since there is a travis.yml in this repository it should be possible, right?

**Comment by sergiomb2** [comment]
not me , don't don't like snaps and appimage I don't know what is it

**Comment by esbante** [comment]
It is the same thing basically. I am not a fan of snaps either, but I was just thinking, as it seems difficult to get full functionality in more modern distributions, it would be easier to package it as an appimage so it can be used independent of distribution and retain the functionality of rawstudio in 14.04 (where it worked best, with enfuse working etc). I will try to do this if I can then, and get back if it works. It is not ideal, but atleast a way to keep rawstudio alive.
